packages/ckeditor5-ckbox/src/ckboximageedit/ckboximageeditcommand.ts

```diff
@@ -0,0 +1,337 @@
+
+		global.document.body.appendChild( this._wrapper );
+
+		const imageElement = this.editor.model.document.selection.getSelectedElement()!;
+		const ckboxImageId = imageElement.getAttribute( 'ckboxImageId' ) as string;
+
+		const processingState: ProcessingState = {
+			ckboxImageId,
+			element: imageElement,
+			controller: new AbortController()
+		};
+
+		window.CKBox.mountImageEditor( this._wrapper, this._prepareOptions( processingState ) );
+
+		this.refresh();
+	}
+
+	public override destroy(): void {
+		this._handleImageEditorOnClose();
+
+		for ( const state of this._processInProgress.values() ) {
+			state.controller.abort();
+		}
+
+		super.destroy();
+	}
+
+	/**
+	 * Indicates if the CKBox Image Editor dialog is already opened.
+	 */
+	private _getValue(): boolean {
+		return this._wrapper !== null;
+	}
+
+	/**
+	 * Creates the options object for the CKBox Image Editor dialog.
+	 */
+	private _prepareOptions( state: ProcessingState ) {
+		const editor = this.editor;
+		const ckboxConfig = editor.config.get( 'ckbox' )!;
+
+		return {
+			assetId: state.ckboxImageId,
+			imageEditing: {
+				allowOverwrite: false
+			},
+			tokenUrl: ckboxConfig.tokenUrl,
+			onClose: () => this._handleImageEditorOnClose(),
+			onSave: ( asset: CKBoxRawAssetDefinition ) => this._handleImageEditorOnSave( state, asset )
+		};
+	}
+
+	/**
+	 * Initializes various event listeners for the `ckboxImageEditor:*` events,
+	 * because all functionality of the `ckboxImageEditor` command is event-based.
+	 */
+	private _prepareListeners(): void {
+		// Abort editing processing when the image has been removed.
+		this.listenTo( this.editor.model.document, 'change:data', () => {
+			const processingStates = this._getProcessingStatesOfDeletedImages();
+
+			processingStates.forEach( processingState => {
+				processingState.controller.abort();
+			} );
+		} );
+	}
+
+	/**
+	 * Gets processing states of images that have been deleted in the mean time.
+	 */
+	private _getProcessingStatesOfDeletedImages(): Array<ProcessingState> {
+		const states: Array<ProcessingState> = [];
+
+		for ( const state of this._processInProgress.values() ) {
+			if ( state.element.root.rootName == '$graveyard' ) {
+				states.push( state );
+			}
+		}
+
+		return states;
+	}
+
+	/**
+	 * Close image editor.
+	 */
+	private _handleImageEditorOnClose() {
+		if ( !this._wrapper ) {
+			return;
+		}
+
+		this._wrapper.remove();
+		this._wrapper = null;
+
+		this.editor.editing.view.focus();
+
+		this.refresh();
+	}
+
+	/**
+	 * Save edited image. In case server respond with "success" replace with edited image,
+	 * otherwise show notification error.
+	 */
+	private _handleImageEditorOnSave( state: ProcessingState, asset: CKBoxRawAssetDefinition ) {
+		const t = this.editor.locale.t;
+		const notification = this.editor.plugins.get( Notification );
+		const pendingActions = this.editor.plugins.get( PendingActions );
```

Comment:
The `Notification` plugin should be included in the `CKBoxImageEditEditing` requires block.